src/Soil-Core/SoilBehaviorRegistry.class.st

```diff
@@ -16,24 +16,29 @@ SoilBehaviorRegistry >> addSpecialObjects [
 		put: SOBehaviorDescription metaId
 ]
 
+{ #category : #'as yet unclassified' }
+SoilBehaviorRegistry >> behaviorDescriptionWithIndex: behaviorIndex andVersion: version transaction: transaction [
+	| versionsOfBehavior |
+
+	self loadHistoryForBehaviorWithId: (SOObjectId segment: 0 index: behaviorIndex) transaction: transaction.
```

Comment:
metaAt: had the purpose that if the segement is known there is no need to create an object id all the time. We need to make a distinction about **id** being the objecId (maybe call it always like that). The other is the **index** which is the index within a segment if the segment is known which is always the case for segment zero

Reply:
I changed it to not create an object id to read the behavior historu